synthtool/gcp/templates/metadata.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import re
+import yaml
+from pathlib import Path
+from typing import Dict
+
+import requests
+
+_RE_SAMPLE_COMMENT_START = r"\[START \w+_quickstart\w*]"
+_RE_SAMPLE_COMMENT_END = r"\[END \w+_quickstart\w*]"
+
+
+class Metadata:
+    def __init__(self, initial_data: dict = {}):
+        self._data = initial_data if initial_data else {}
```

Comment:
I like how this code has been moved into a separate .py file, but If self._data is the only private data, then I actually prefer the functional structure of the original code.